test/unit/DCAStrategies/dca-strategies-positions-handler.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { DCAStrategiesPositionsHandlerMock__factory, DCAStrategiesPositionsHandlerMock, IERC20 } from '@typechained';
+import { FakeContract, smock } from '@defi-wonderland/smock';
+import { constants } from '@test-utils';
+import { given, then, when, contract } from '@test-utils/bdd';
+import { snapshot } from '@test-utils/evm';
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import { BigNumber } from '@ethersproject/bignumber';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+
+contract('DCAStrategiesPositionsHandler', () => {
+  let snapshotId: string;
+  let DCAStrategiesPositionsHandlerMock: DCAStrategiesPositionsHandlerMock;
+  let user: SignerWithAddress, random: SignerWithAddress, governor: SignerWithAddress;
+  let factory: DCAStrategiesPositionsHandlerMock__factory;
+  let tokenA: FakeContract<IERC20>, tokenB: FakeContract<IERC20>;
+  let SHARE_TOKEN_A;
+  let SHARE_TOKEN_B;
+  let SHARES: any[];
+
+  before('Setup accounts and contracts', async () => {
+    [user, random, governor] = await ethers.getSigners();
+    factory = await ethers.getContractFactory('DCAStrategiesPositionsHandlerMock');
+    DCAStrategiesPositionsHandlerMock = await factory.deploy();
+    snapshotId = await snapshot.take();
+    tokenA = await smock.fake('IERC20');
+    tokenB = await smock.fake('IERC20');
+    SHARE_TOKEN_A = { token: tokenA.address, share: BigNumber.from(50e2) };
+    SHARE_TOKEN_B = { token: tokenB.address, share: BigNumber.from(50e2) };
+    SHARES = [SHARE_TOKEN_A, SHARE_TOKEN_B];
+  });
+
+  beforeEach(async () => {
+    await snapshot.revert(snapshotId);
+  });
+
+  describe('deposit', () => {
+    let tx: TransactionResponse;
+    let toDeposit: BigNumber = ethers.utils.parseUnits('300');
+    let amountOfSwaps: number = 5;
+    let swapInterval: number = 7 * 24 * 60 * 60; // 1 week
```

Comment:
Just as an fyi, when you assign a value, typescript can infer the type
```suggestion
    let amountOfSwaps = 5;
    let swapInterval = 7 * 24 * 60 * 60; // 1 week
```